keyboards/planck/keymaps/oeywil/keymap.c

```diff
@@ -116,81 +143,258 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
-   * | Ctrl |      |  Alt |      |    Space    |    Enter    |      |      | Def  | Steam|
+   * | Ctrl |      |      |  Alt |      | PlyPa| Next |      |      | Mute | VolD | VolUp|
    * `-----------------------------------------------------------------------------------'
    */
-  [_GAMERAISE] = LAYOUT_planck_grid(
-    KC_TRNS, KC_1,  KC_2,    KC_3,    KC_4,  KC_5,    KC_NO, KC_NO,   KC_NO, KC_NO, KC_NO,   KC_VOLU,
-    KC_TRNS, KC_F1, KC_F2,   KC_F3,   KC_F4, KC_F5,   KC_NO, KC_NO,   KC_NO, KC_NO, KC_NO,   KC_VOLD,
-    KC_TRNS, KC_NO, KC_NO,   KC_NO,   KC_NO, KC_NO,   KC_NO, KC_NO,   KC_NO, KC_NO, KC_NO,   KC_MUTE,
-    KC_TRNS, KC_NO, KC_TRNS, KC_TRNS, KC_NO, KC_TRNS, KC_NO, KC_TRNS, KC_NO, KC_NO, GAMEOFF, STEAM
+  [_GLOW] = LAYOUT_planck_grid(
+    KC_TRNS, KC_1,    KC_2,    KC_3,    KC_NO,   KC_NO,   KC_NO,   KC_NO, KC_NO,   KC_NO,   KC_NO,   LSFT(KC_F7),
+    KC_TRNS, KC_4,    KC_5,    KC_6,    KC_NO,   KC_NO,   KC_NO,   KC_NO, KC_NO,   KC_NO,   KC_NO,   KC_F12,
+    KC_TRNS, KC_7,    KC_8,    KC_9,    KC_0,    KC_NO,   KC_NO,   KC_NO, KC_NO,   KC_NO,   KC_NO,   KC_MUTE,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_MPLY, KC_MNXT, KC_NO, KC_TRNS, KC_MUTE, KC_VOLD, KC_VOLU
   )
 };
+
+// tap dance declarations
+int cur_dance (qk_tap_dance_state_t *state) {
+  if (state->count == 1) {
+    if (state->interrupted || !state->pressed)  return SINGLE_TAP;
+    else return SINGLE_HOLD;
+  }
+  else if (state->count == 2) {
+    if (state->interrupted) return DOUBLE_SINGLE_TAP;
+    else if (state->pressed) return DOUBLE_HOLD;
+    else return DOUBLE_TAP;
+  }
+  if (state->count == 3) {
+    if (state->interrupted || !state->pressed)  return TRIPLE_TAP;
+    else return TRIPLE_HOLD;
+  }
+  else return 8;
+}
+
+static tap xtap_state = {
+  .is_press_action = true,
+  .state = 0
+};
+
+void lsft_finished (qk_tap_dance_state_t *state, void *user_data) {
+  xtap_state.state = cur_dance(state);
+  switch (xtap_state.state) {
+    case SINGLE_TAP:
+      register_mods(MOD_BIT(KC_LSFT));
+      register_code(KC_8);
+      break;
+    case SINGLE_HOLD:
+      register_code(KC_LSFT);
+      break;
+    case DOUBLE_TAP:
+      register_code(KC_NUBS);
+      break;
+    case DOUBLE_SINGLE_TAP:
+      register_code(KC_NUBS);
+      break;
+  }
+}
+
+void lsft_reset (qk_tap_dance_state_t *state, void *user_data) {
+  switch (xtap_state.state) {
+    case SINGLE_TAP:
+      unregister_code(KC_8);
+      unregister_mods(MOD_BIT(KC_LSFT));
+      break;
+    case SINGLE_HOLD:
+      unregister_code(KC_LSFT);
+      break;
+    case DOUBLE_TAP:
+      unregister_code(KC_NUBS);
+      break;
+    case DOUBLE_SINGLE_TAP:
+      unregister_code(KC_NUBS);
+      break;
+  }
+  xtap_state.state = 0;
+}
+
+void rsft_finished (qk_tap_dance_state_t *state, void *user_data) {
+  xtap_state.state = cur_dance(state);
+  switch (xtap_state.state) {
+    case SINGLE_TAP:
+      register_mods(MOD_BIT(KC_LSFT));
```

Comment:
Actually, you could do `register_code16(S(KC_9));` here instead of the register mod and code functions.   Same with `unregister_code16`

Reply:
@oeywil did you want to change these, or were you fine with how it is?

Reply:
I’ll update them soon.

Reply:
Fixed in [04d54b9](https://github.com/qmk/qmk_firmware/pull/5614/commits/04d54b92f308952ee992d0e45b15076fc2369f6c). register_code16() doesn't work with RALT or ALGR modifier so had to leave lctl and rctl functions unchanged. Possible bug?

Reply:
Yeah, that would be a bug.  It should work with them. :(

Reply:
Just tested again. register_code16(RALT(KC_7)) and register_code16(ALGR(KC_7)) does nothing. Works with register_code and register_mods. RALT + KC_7 should give {